modules/services/syncthing.nix

```diff
@@ -8,27 +8,53 @@ with lib;
   options = {
     services.syncthing = {
       enable = mkEnableOption "Syncthing continuous file synchronization";
+
+      tray = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable QSyncthingTray service.";
+      };
     };
   };
 
-  config = mkIf config.services.syncthing.enable {
-    systemd.user.services.syncthing = {
-      Unit = {
-        Description = "Syncthing - Open Source Continuous File Synchronization";
-        Documentation = "man:syncthing(1)";
-        After = [ "network.target" ];
-      };
+  config = mkIf config.services.syncthing.enable (mkMerge [
```

Comment:
I thinks it should be possible to avoid `mkMerge` in this case. Something like this could work:

```nix
  config = mkIf config.services.syncthing.enable {
    systemd.user.services = {
      syncthing = {
        Unit = {
          Description = "Syncthing - Open Source Continuous File Synchronization";
          Documentation = "man:syncthing(1)";
          After = [ "network.target" ];
        };

        Service = {
          ExecStart = "${pkgs.syncthing}/bin/syncthing -no-browser -no-restart -logflags=0";
          Restart = "on-failure";
          SuccessExitStatus = [ 3 4 ];
          RestartForceExitStatus = [ 3 4 ];
        };

        Install = {
          WantedBy = [ "default.target" ];
        };
      };

      qsyncthingtray = mkIf config.services.syncthing.tray {
        Unit = {
          Description = "QSyncthingTray";
          After = [ "graphical-session-pre.target" ];
          PartOf = [ "graphical-session.target" ];
        };

        Install = {
          WantedBy = [ "graphical-session.target" ];
        };

        Service = {
          ExecStart = "${pkgs.qsyncthingtray}/bin/QSyncthingTray";
        };
      };
    };
  };
```

Reply:
Good point. But for some reason using this code I get "error: value is a string while a set was expected" error. The problem seems to be in this condition: `qsyncthingtray = mkIf config.services.syncthing.tray`. Removing it fixes the problem.

Reply:
Ah, sorry! Yeah, that's right. This won't work because `systemd.user.services` is not a submodule, just a plain attribute set.